src/ipc-handlers.ts

```diff
@@ -238,48 +241,63 @@ export async function connectWpcomSite(
 
 	const connections = userData.connectedWpcomSites[ currentUserId ];
 
-	sites.forEach( ( siteToAdd ) => {
-		const isAlreadyConnected = connections.some(
-			( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
-		);
+	list.forEach( ( { sites, localSiteId } ) => {
+		sites.forEach( ( siteToAdd ) => {
+			const isAlreadyConnected = connections.some(
+				( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
+			);
 
-		// Add the site if it's not already connected
-		if ( ! isAlreadyConnected ) {
-			connections.push( { ...siteToAdd, localSiteId } );
-		}
+			// Add the site if it's not already connected
+			if ( ! isAlreadyConnected ) {
+				connections.push( { ...siteToAdd, localSiteId } );
+			}
+		} );
 	} );
 
 	await saveUserData( userData );
 
-	return connections.filter( ( conn ) => conn.localSiteId === localSiteId );
+	return connections.filter( ( conn ) =>
+		list.some( ( { localSiteId } ) => conn.localSiteId === localSiteId )
+	);
 }
 
+type DataDisconnectWpcomSite = { siteIds: number[]; localSiteId: string };
+
 export async function disconnectWpcomSite(
 	event: IpcMainInvokeEvent,
-	siteIds: number[],
-	localSiteId: string
+	data: DataDisconnectWpcomSite | DataDisconnectWpcomSite[]
 ) {
+	const list = Array.isArray( data ) ? data : [ data ];
+
 	const userData = await loadUserData();
 	const currentUserId = userData.authToken?.id;
 
 	if ( ! currentUserId ) {
 		throw new Error( 'User not authenticated' );
 	}
 
-	const connections = userData.connectedWpcomSites?.[ currentUserId ] || [];
+	const connectedWpcomSites = userData.connectedWpcomSites;
 
-	const updatedConnections = connections.filter(
-		( conn ) => ! ( siteIds.includes( conn.id ) && conn.localSiteId === localSiteId )
-	);
+	// Totally unreal case, added it to make types below more clear and if this error happens, we definitely have something wrong
+	if ( ! connectedWpcomSites || ! connectedWpcomSites?.[ currentUserId ]?.length ) {
+		throw new Error(
+			'Something went wrong, since you are trying to disconnect something, but there are no stored connections yet'
+		);
+	}
```

Comment:
If `data` doesn't match any of the connected sites, we just silently fail. `connectedWpcomSites` being undefined could potentially indicate a more serious error (I'm not sure exactly what, though). However, if it's simply empty, I would be OK with also just silently failing.

Reply:
The main point is helping TS to understand the code and for us to simplify maintenance in the future:
1) The first condition is not critical
2) But the second one (`.?` optional chaining) - make the return type of function as `SyncSite[] | undefined`, so I didn't want to have `undefined` as possible return type and adding conditions in places where this function is used.
Ofc, we can do something like this:
```
return (connectedWpcomSites || []).[ currentUserId ].filter( ( conn ) =>
		list.some( ( { localSiteId } ) => conn.localSiteId === localSiteId )
	);
```
But in this case, another developer (who will be working with this code in the future) could wonder - "how `connectedWpcomSites` can be `undefiend`". So it can confuse. But, thinking about it right now one more time, maybe `connectedWpcomSites || []` also good idea.
WDYT about my thoughts?

<img width="654" alt="Screenshot 2024-11-29 at 13 31 33" src="https://github.com/user-attachments/assets/e3933a3c-df71-4181-8fe0-21768bc6253e">

Reply:
```suggestion
	if ( ! Array.isArray( connectedWpcomSites?.[ currentUserId ] ) ) {
		throw new Error(
			'Something went wrong, since you are trying to disconnect something, but there are no stored connections yet'
		);
	}
```
Ultimately this is kind of nitpicky, so we don't need to dwell on it too much, but this suggestion seems to also account for your point about always returning an array.

Reply:
I am ok with any approach to help TS understand what is going on inside this function 😄 I was looking for the best (between the worst) option, and the current one was looking good.
I checked your suggestion and TS is happy with it too, and as for me even simpler to read than mine, so I will apply it in the next commit, thanks 🙂